### PR TITLE
update(qos): support anonymous access.

### LINKED ISSUE #11015: Support optional opening of some qos commands to the external network
- [ ] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.

## Describe the proposal

Related with https://github.com/apache/dubbo/issues/11014

Background: Dubbo QoS supports docking with kubernetes probes, but foreignIp needs to be enabled. Due to the high-risk instructions in qos, this will lead to a series of security problems.

We should distinguish those commands with low-risk and high-risk, and support all client to request low-risk command for healthy check or metrics collection purpose.

### PR DESCRIPTION
**Pull request by Koooooo-7**
We should distinguish those commands with low-risk and high-risk, and support all client to request low-risk command for healthy check or metrics collection purpose.

**TODO**:
-  Indicate the permission level for each cmd (currently marked `live`, `startup`, `ready`,`quit` as public and others `PROTECTED`).
-  Docs.

close #11015 

## Brief changelog

Add new config `qos.anonymous.access.permission.level` to enable any foreign ip access. Default is `Cmd.PermissionLevel.NONE`, can not access any cmd (equivalent to disable the anonymous access config).

Permission level has 4 levels, `PUBLIC(1)`, `PROTECTED(2)`,`PRIVATED(3)` and `NONE`( reversed to default anonymous permission level).

It allows to access all the cmds which have required permission is equal or lower than config permission level. i.e.
If `qos.anonymous.access.permission.level=PROTECTED`, it can anonymous access all the `PUBLIC` and `PROTECTED` cmds.

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by Koooooo-7** [comment]
Hi @AlbumenJ , PTAL.

**Review by AlbumenJ — CHANGES_REQUESTED** [review]
Please rebase this PR onto `3.2` branch.

File: dubbo-plugin/dubbo-qos/src/main/java/org/apache/dubbo/qos/command/annotation/Cmd.java [hunk 1]
```diff
@@ -55,4 +58,57 @@
...
+        PRIVATE(3),
+
+        /**
+         * It is the reserved default anonymous permission level, can not access any command
+         */
+        NONE(Integer.MIN_VALUE),
+
+        ;
+        private final int level;
+
```
Comment (AlbumenJ) [inline dubbo-plugin/dubbo-qos/src/main/java/org/apache/dubbo/qos/command/annotation/Cmd.java:91]: Move out as a public class

File: dubbo-plugin/dubbo-qos/src/main/java/org/apache/dubbo/qos/command/DefaultAnonymousAccessPermissionChecker.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import org.apache.dubbo.qos.command.annotation.Cmd;
+
+public class DefaultAnonymousAccessPermissionChecker implements PermissionChecker {
+    public static final DefaultAnonymousAccessPermissionChecker INSTANCE = new DefaultAnonymousAccessPermissionChecker();
+
+    @Override
+    public boolean access(CommandContext commandContext, Cmd.PermissionLevel defaultCmdRequiredPermissionLevel) {
+        return commandContext.hasPermission(defaultCmdRequiredPermissionLevel);
+    }
+}
```
Comment (AlbumenJ) [inline dubbo-plugin/dubbo-qos/src/main/java/org/apache/dubbo/qos/command/DefaultAnonymousAccessPermissionChecker.java:28]: Is it really necessary to create this class separately.

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-plugin/dubbo-qos/src/main/java/org/apache/dubbo/qos/command/annotation/Cmd.java (hunk 1 above)
Comment (Koooooo-7) [reply dubbo-plugin/dubbo-qos/src/main/java/org/apache/dubbo/qos/command/annotation/Cmd.java]: updated.

**Comment by Koooooo-7** [comment]
> Please rebase this PR onto `3.2` branch.

As per the large diff of commits between `3.1` to `3.2` now, It seems hard to rebase the pr to `3.2` , unless making this pr mess with other commits which already in `3.1` (include  `qos` updates). I suggest to merge it in `3.1` and sync to `3.2` later.

File: dubbo-plugin/dubbo-qos/src/main/java/org/apache/dubbo/qos/protocol/QosProtocolWrapper.java [hunk 3]
```diff
@@ -116,6 +118,7 @@ private void startQosServer(URL url) {
             int port = url.getParameter(QOS_PORT, QosConstants.DEFAULT_PORT);
             boolean acceptForeignIp = Boolean.parseBoolean(url.getParameter(ACCEPT_FOREIGN_IP, "false"));
             String acceptForeignIpWhitelist = url.getParameter(ACCEPT_FOREIGN_IP_WHITELIST, StringUtils.EMPTY_STRING);
+            String anonymousAccessPermissionLevel = url.getParameter(ANONYMOUS_ACCESS_PERMISSION_LEVEL, PermissionLevel.NONE.name());
```
Comment (AlbumenJ) [inline dubbo-plugin/dubbo-qos/src/main/java/org/apache/dubbo/qos/protocol/QosProtocolWrapper.java:121]: Default level could be public.

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-plugin/dubbo-qos/src/main/java/org/apache/dubbo/qos/protocol/QosProtocolWrapper.java (hunk 3 above)
Comment (Koooooo-7) [reply dubbo-plugin/dubbo-qos/src/main/java/org/apache/dubbo/qos/protocol/QosProtocolWrapper.java]: updated.

**Comment by Koooooo-7** [comment]
It seems there has IT issue same to branch `3.2` latest commit.

**Review by AlbumenJ — APPROVED** [review]
LGTM. Would you please add some document for this PR in dubbo-website.

**Pull request merged**